jira.go

```diff
@@ -63,46 +63,59 @@ func getJiraTickets(endpointAPI string, orgID string, projectID string, token st
 
-		jiraTicket.Fields.Projects.ID = jiraProjectID
-		jiraTicket.Fields.IssueTypes.Name = jiraTicketType
-		jiraTicket.Fields.Assignees.ID = assigneeID
-		if labels != "" {
-			jiraTicket.Fields.Labels = strings.Split(labels, ",")
-		}
-
-		if priorityIsSeverity {
-			var priority PriorityType
-			jiraMappingEnvVarName := fmt.Sprintf("SNYK_JIRA_PRIORITY_FOR_%s_VULN", strings.ToUpper(jsonVuln.K("severity").String().Value))
-			val, present := os.LookupEnv(jiraMappingEnvVarName)
-			if present {
-				priority.Name = val
+	jiraTicket.Fields.Projects.ID = jiraProjectID
+	jiraTicket.Fields.IssueTypes.Name = jiraTicketType
+	jiraTicket.Fields.Assignees.ID = assigneeID
+	if labels != "" {
+		jiraTicket.Fields.Labels = strings.Split(labels, ",")
+	}
+	if priorityIsSeverity {
+		var priority PriorityType
+		jiraMappingEnvVarName := fmt.Sprintf("SNYK_JIRA_PRIORITY_FOR_%s_VULN", strings.ToUpper(jsonVuln.K("severity").String().Value))
+		val, present := os.LookupEnv(jiraMappingEnvVarName)
+		if present {
+			priority.Name = val
+		} else {
+			if jsonVuln.K("severity").String().Value == "critical" {
+				priority.Name = "Highest"
 			} else {
-				if jsonVuln.K("severity").String().Value == "critical" {
-					priority.Name = "Highest"
-				} else {
 
-					priority.Name = strings.Title(jsonVuln.K("severity").String().Value)
+				priority.Name = strings.Title(jsonVuln.K("severity").String().Value)
 
-				}
 			}
-			jiraTicket.Fields.Priority = &priority
-		}
 
-		ticket, err := json.Marshal(jiraTicket)
-		if err != nil {
-			log.Fatalln(err)
 		}
+		jiraTicket.Fields.Priority = &priority
+	}
 
-		responseData := makeSnykAPIRequest("POST", endpointAPI+"/v1/org/"+orgID+"/project/"+projectInfo.K("id").String().Value+"/issue/"+vulnID+"/jira-issue", token, ticket)
-		responseDataAggregated += "\n" + string(responseData)
+	ticket, err := json.Marshal(jiraTicket)
+	if err != nil {
+		log.Fatalln(err)
 	}
-	return responseDataAggregated
+
+	responseData := makeSnykAPIRequest("POST", endpointAPI+"/v1/org/"+orgID+"/project/"+projectInfo.K("id").String().Value+"/issue/"+vulnID+"/jira-issue", token, ticket)
+
+	return responseData
+}
+
+func openJiraTickets(endpointAPI string, orgID string, token string, jiraProjectID string, jiraTicketType string, assigneeID string, labels string, projectInfo jsn.Json, vulnsForJira map[string]interface{}, priorityIsSeverity bool) string {
+	fullResponseDataAggregated := ""
+
+	for _, vulnForJira := range vulnsForJira {
+		responseDataAggregatedByte := openJiraTicket(endpointAPI, orgID, token, jiraProjectID, jiraTicketType, assigneeID, labels, projectInfo, vulnForJira, priorityIsSeverity)
+
+		if strings.Contains(string(responseDataAggregatedByte), "Field 'priority'") && priorityIsSeverity == true {
+			fmt.Println("Retrying with priorityIsSeverity set to false, max retry 1")
+			responseDataAggregatedByte = openJiraTicket(endpointAPI, orgID, token, jiraProjectID, jiraTicketType, assigneeID, labels, projectInfo, vulnForJira, false)
```

Comment:
how do we know we don't end up retrying again and again? if it fails for some other reasons? maybe "Highest" is not the default value after all. Probably should have a retry counter or something as failsafe